### PR TITLE
[Refactor] Rename enum TableType.LAKE to TableType.CLOUD_NATIVE

### PR DESCRIPTION
**Pull request by sduzh**
Also rename TableType.LAKE_MATERIALIZED_VIEW to TableType.CLOUD_NATIVE_MATERIALIZED_VIEW

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.0

**Review by kevincai — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/alter/Alter.java [hunk 1]
```diff
@@ -557,7 +557,7 @@ public void processAlterTable(AlterTableStmt stmt) throws UserException {
                 ErrorReport.reportDdlException(ErrorCode.ERR_BAD_TABLE_ERROR, tableName);
             }
 
-            if (!table.isOlapOrLakeTable()) {
+            if (!table.isOlapOrCloudNativeTable()) {
                 throw new DdlException("Do not support alter non-OLAP or non-LAKE table[" + tableName + "]");
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/alter/Alter.java:561]: update the message

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java [hunk 2]
```diff
@@ -520,7 +520,7 @@ private boolean needToRefreshNonPartitionTable(Table partitionTable) {
     }
 
     private boolean supportRefreshByPartition(Table table) {
-        if (table.isLocalTable() || table.isHiveTable() || table.isLakeTable()) {
+        if (table.isLocalTable() || table.isHiveTable() || table.isCloudNativeTable()) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:523]:
```suggestion
        if (table.isLocalTable() || table.isHiveTable() || table. isCloudNativeTableOrMaterializedView()) {
```

Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:523]: maybe a bug

File: fe/fe-core/src/main/java/com/starrocks/catalog/Table.java [hunk 3]
```diff
@@ -227,24 +264,24 @@ public boolean isLocalTable() {
...
+    public boolean isCloudNativeTableOrMaterializedView() {
+        return isCloudNativeTable() || isCloudNativeMaterializedView();
     }
 
     public boolean isMaterializedView() {
-        return isOlapMaterializedView() || isLakeMaterializedView();
+        return isOlapMaterializedView() || isCloudNativeMaterializedView();
     }
 
     public boolean isNativeTable() {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Table.java:283]:
```suggestion
    public boolean isNativeTableOrMaterializedView() {
```

File: fe/fe-core/src/main/java/com/starrocks/catalog/Table.java [hunk 4]
```diff
@@ -227,24 +264,24 @@ public boolean isLocalTable() {
...
-        return isLocalTable() || isCloudNativeTable();
+        return isLocalTable() || isCloudNativeTableOrMaterializedView();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Table.java:284]:
```suggestion
        return isLocalTableOrMaterializedView()() || isCloudNativeTableOrMaterializedView();
```

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 2 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:523]: or `if (table.isNativeTableOrMaterializedView() || table.isHiveTable())`

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:523]: IMO we should fix this in a separate patch.

File: fe/fe-core/src/main/java/com/starrocks/catalog/Table.java [hunk 5]
```diff
@@ -78,23 +78,60 @@ public class Table extends MetaObject implements Writable, GsonPostProcessable {
...
+        @SerializedName("HUDI")
         HUDI,
+        @SerializedName("JDBC")
         JDBC,
+        @SerializedName("MATERIALIZED_VIEW")
         MATERIALIZED_VIEW,
-        LAKE,
+        @SerializedName("LAKE") // for backward and rollback compatibility
+        CLOUD_NATIVE,
+        @SerializedName("DELTALAKE")
```
[the remark this answers is not in the document]
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Table.java:109]: Updated

File: fe/fe-core/src/main/java/com/starrocks/alter/Alter.java (hunk 1 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/alter/Alter.java]: updated

File: fe/fe-core/src/main/java/com/starrocks/catalog/Table.java (hunk 3 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Table.java]: updated

**Review by wyb — DISMISSED** [review]

**Review by imay — DISMISSED** [review]

**Comment by sduzh** [comment]
Fixed failed test case, PTAL @imay @wyb

**Review by imay — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**